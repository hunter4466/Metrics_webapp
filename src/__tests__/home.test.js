import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Home from '../components/home';
import store from '../redux/configureStore';

it('Renders correctly', () => {
  const tree = renderer.create(<Provider store={store}><Home /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Home component', () => {
  test("Rendererd component contains 'STATS BY COMPANY'", () => {
    render(<Provider store={store}><Home /></Provider>);
    expect(screen.getByText(/STATS BY COMPANY/i)).toBeInTheDocument();
  });
  test("Rendererd component contains 'Cripto Up to date'", () => {
    render(<Provider store={store}><Home /></Provider>);
    expect(screen.getByText(/Cripto Up to date/i)).toBeInTheDocument();
  });
});
