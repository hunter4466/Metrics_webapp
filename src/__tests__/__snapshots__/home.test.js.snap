// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <div
    className="Home_Header_1"
  >
    <h1>
      Cripto Up to date
    </h1>
    <h2>
      12 results
    </h2>
  </div>
  <div
    className="Home_Header_2"
  >
    <h1>
      STATS BY COMPANY
    </h1>
  </div>
  <ul
    className="home_btns_cont"
  />
</div>
`;
